server/internal/service/slack.go

```diff
@@ -46,3 +61,76 @@ func (svc *SlackService) GetSlackConfig() (*responseparams.SlackConfigResponse,
 		VerificationToken: config.Config.Slack.VerificationToken,
 	}, nil
 }
+
+// CreateCustomer create customer if not exists.
+func (svc *SlackService) CreateCustomer(userId string, profile *slack.UserProfile) (string, uint, error) {
+	// check if exists
+	temp, err := svc.CustomerSlackDao.FindFirstByUUID(userId)
+	if temp != nil || err == nil {
+		return temp.UUID, temp.Id, err
+	}
+
+	// Set unique email
+	if profile.Email == "" {
+		profile.Email = fmt.Sprintf("%s %s", "NOEMAIL", utils.GenUUIDWithoutHyphen())
+	}
+
+	customerSlack := customerroles.CustomerSlack{
+		UserBase: base.UserBase{
+			// UUID:        fmt.Sprintf("%s%s", SLACK_PERFIX, userId),
+			UUID:        userId,
+			Email:       profile.Email,
+			Nickname:    profile.FirstName,
+			Avatar:      profile.Image512,
+			Description: profile.Title,
+			IsEnable:    true,
+		},
+		FirstName:             profile.FirstName,
+		LastName:              profile.LastName,
+		RealName:              profile.RealName,
+		RealNameNormalized:    profile.RealNameNormalized,
+		DisplayName:           profile.DisplayName,
+		DisplayNameNormalized: profile.DisplayNameNormalized,
+		Skype:                 profile.Skype,
+		Phone:                 profile.Phone,
+		Image24:               profile.Image24,
+		Image32:               profile.Image32,
+		Image48:               profile.Image48,
+		Image72:               profile.Image72,
+		Image192:              profile.Image192,
+		Image512:              profile.Image512,
+		ImageOriginal:         profile.ImageOriginal,
```

Comment:
the image number here is pixel?

Reply:
Yes, these fields are from github.com/slack-go/slack@v0.12.1/users.go:UserProfile, I save them to local OpenKF storage.